src/main/type.ts

```diff
@@ -0,0 +1,299 @@
+import type { BinaryLike } from 'crypto'
+import type { IncomingMessage } from 'http'
+import type { Readable as ReadableStream } from 'stream'
+
+import type { Client } from './minio.d.mts'
+
+export type Binary = string | Buffer
+export type RequestHeader = Record<string, string | boolean | number>
+
+export interface IRequest {
+  protocol: string
+  port?: number | string
+  method: string
+  path: string
+  headers: RequestHeader
+}
+
+export type ICanonicalRequest = string
+
+export interface ICredentials {
+  accessKey: string
+  secretKey: string
+  sessionToken?: string
+}
+
+type ResponseCallback = (err: unknown, res: IncomingMessage) => void
+
+export type UploadID = string
+
+export type LockUnit = 'Days' | 'Years'
+export type LegalHoldStatus = 'ON' | 'OFF'
+export type NoResultCallback = (error: unknown | null) => void
+export type ResultCallback<T> = (error: unknown | null, result: T) => void
+export type TagList = Record<string, string>
+export type EmptyObject = Record<string, never>
+export type VersionIdentificator = Pick<RetentionOptions, 'versionId'>
+export type Lifecycle = LifecycleConfig | null | ''
+export type Lock = LockConfig | EmptyObject
+export type Encryption = EncryptionConfig | EmptyObject
+export type Retention = RetentionOptions | EmptyObject
+export type IsoDate = string
+
+export interface GetObjectOpt {
```

Comment:
I prefer using "GetObjectOptions" for naming things. 

but this should work fine if not directly exported to users.